aptos-move/move-examples/whitelist/sources/mint.move

```diff
@@ -0,0 +1,273 @@
+
+        // store next GUID to derive object address later
+        let token_creation_num = account::get_guid_next_creation_num(@whitelist_example);
+
+        // generate next token name based on current collection supply
+        let token_name = next_token_name_from_supply(
+            signer::address_of(resource_signer),
+            string::utf8(BASE_TOKEN_NAME),
+            string::utf8(COLLECTION_NAME),
+        );
+
+        // mint token and send it to the receiver
+        aptos_token::mint(
+            resource_signer,
+            string::utf8(COLLECTION_NAME),
+            string::utf8(TOKEN_DESCRIPTION),
+            token_name,
+            string::utf8(TOKEN_URI),
+            vector<String> [ ],
+            vector<String> [ ],
+            vector<vector<u8>> [ ],
+        );
+        let token_object = object::address_to_object<AptosToken>(object::create_guid_object_address(@whitelist_example, token_creation_num));
+        object::transfer(resource_signer, token_object, signer::address_of(receiver));
+    }
+
+    /// generates the next token name by concatenating the supply onto the base token name
+    fun next_token_name_from_supply(
+        creator_address: address,
+        base_token_name: String,
+        collection_name: String,
+    ): String {
+        let collection_addr = collection::create_collection_address(&creator_address, &collection_name);
+        let collection_object = object::address_to_object<Collection>(collection_addr);
+        let current_supply = option::borrow(&collection::count(collection_object));
+        let format_string = base_token_name;
+        // if base_token_name == Token Name
+        string::append_utf8(&mut format_string, b" #{}");
+        // 'Token Name #1' when supply == 0
+        string_utils::format1(string::bytes(&format_string), *current_supply + 1)
+    }
+
+    public entry fun add_addresses_to_tier(
+        owner: &signer,
+        tier_name: String,
+        addresses: vector<address>,
+    ) acquires MintConfiguration {
+        assert!(signer::address_of(owner) == @owner, error::permission_denied(ENOT_AUTHORIZED));
+        let mint_configuration = borrow_global_mut<MintConfiguration>(@whitelist_example);
+        let resource_signer = &account::create_signer_with_capability(&mint_configuration.signer_capability);
+        whitelist::add_addresses_to_tier(resource_signer, tier_name, addresses);
+    }
+
+    public entry fun remove_addresses_from_tier(
+        owner: &signer,
+        tier_name: String,
+        addresses: vector<address>,
+    ) acquires MintConfiguration {
+        assert!(signer::address_of(owner) == @owner, error::permission_denied(ENOT_AUTHORIZED));
+        let mint_configuration = borrow_global_mut<MintConfiguration>(@whitelist_example);
+        let resource_signer = &account::create_signer_with_capability(&mint_configuration.signer_capability);
+        whitelist::remove_addresses_from_tier(resource_signer, tier_name, addresses);
+    }
```

Comment:
Also, could you add a function to clean the whitelist object since it is ephemeral.

Reply:
Yea, will do!